contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -113,8 +115,10 @@ pub fn call_contract(
         token: token.clone(),
     };
 
-    let res = match determine_routing_destination(&client, &msg.destination_chain)? {
-        RoutingDestination::Nexus => route_to_nexus(&client, nexus, msg, token)?,
+    let res = match determine_routing_destination(&client, &msg.destination_chain, &chain_name)? {
+        RoutingDestination::Nexus => {
+            Response::new().add_messages(route_to_nexus(&client, &nexus, msg, token)?)
+        }
         RoutingDestination::Router if token.is_none() => {
```

Comment:
```suggestion
        RoutingDestination::Router | RoutingDestination::Axelarnet if token.is_none() => {
```
routing message to self chain should work as well. add a test as well?